File: face_api_demo/utils/timezone_helper.py
"""
Timezone Utilities for UTC+7 (Indochina Time / Bangkok Time)
"""

from datetime import datetime, timezone, timedelta
import pytz

# Define UTC+7 timezone
ICT = pytz.timezone('Asia/Bangkok')  # UTC+7
UTC_OFFSET = timedelta(hours=7)


def get_now() -> datetime:
    """
    Get current date/time in UTC+7
    
    Returns:
        datetime: Current time in UTC+7 timezone
    """
    return datetime.now(ICT)


def to_local_time(utc_dt: datetime) -> datetime:
    """
    Convert UTC datetime to UTC+7
    
    Args:
        utc_dt: UTC datetime object
        
    Returns:
        datetime: Datetime in UTC+7 timezone
    """
    if utc_dt.tzinfo is None:
        utc_dt = pytz.utc.localize(utc_dt)
    return utc_dt.astimezone(ICT)


def to_utc(local_dt: datetime) -> datetime:
    """
    Convert UTC+7 datetime to UTC
    
    Args:
        local_dt: Datetime in UTC+7 timezone
        
    Returns:
        datetime: UTC datetime object
    """
    if local_dt.tzinfo is None:
        local_dt = ICT.localize(local_dt)
    return local_dt.astimezone(pytz.utc)


def get_utc_now_for_storage() -> datetime:
    """
    Get current UTC time for database storage
    Returns UTC time but represents current ICT moment
    
    Returns:
        datetime: Current time in UTC (for database storage)
    """
    return datetime.now(pytz.utc)


def format_datetime(dt: datetime, fmt: str = '%Y-%m-%d %H:%M:%S') -> str:
    """
    Format datetime in UTC+7 timezone
    
    Args:
        dt: Datetime object
        fmt: Format string
        
    Returns:
        str: Formatted datetime string
    """
    local_dt = to_local_time(dt) if dt.tzinfo != ICT else dt
    return local_dt.strftime(fmt)


class TimezoneInfo:
    """Timezone information constants"""
    NAME = 'Asia/Bangkok'
    DISPLAY_NAME = 'Indochina Time (ICT)'
    ABBREVIATION = 'ICT'
    OFFSET_HOURS = 7
    TIMEZONE = ICT

File: FaceIdBackend/FaceIdBackend.Shared/Utils/TimezoneHelper.cs
using System;

namespace FaceIdBackend.Shared.Utils
{
    /// <summary>
    /// Timezone utilities for UTC+7 (Indochina Time / Bangkok Time)
    /// </summary>
    public static class TimezoneHelper
    {
        // UTC+7 timezone
        private static readonly TimeZoneInfo IndochinaTimeZone =
            TimeZoneInfo.FindSystemTimeZoneById("SE Asia Standard Time"); // UTC+7

        /// <summary>
        /// Get current date/time in UTC+7
        /// </summary>
        public static DateTime GetNow()
        {
            return TimeZoneInfo.ConvertTimeFromUtc(DateTime.UtcNow, IndochinaTimeZone);
        }

        /// <summary>
        /// Convert UTC to UTC+7
        /// </summary>
        public static DateTime ToLocalTime(DateTime utcDateTime)
        {
            if (utcDateTime.Kind != DateTimeKind.Utc)
            {
                utcDateTime = DateTime.SpecifyKind(utcDateTime, DateTimeKind.Utc);
            }
            return TimeZoneInfo.ConvertTimeFromUtc(utcDateTime, IndochinaTimeZone);
        }

        /// <summary>
        /// Convert UTC+7 to UTC
        /// </summary>
        public static DateTime ToUtc(DateTime localDateTime)
        {
            return TimeZoneInfo.ConvertTimeToUtc(localDateTime, IndochinaTimeZone);
        }

        /// <summary>
        /// Get UTC+7 DateTime for database storage (stored as UTC)
        /// </summary>
        public static DateTime GetUtcNowForStorage()
        {
            // Store UTC in database, but represent current ICT time
            return DateTime.UtcNow;
        }

        /// <summary>
        /// Timezone info for display
        /// </summary>
        public static class Info
        {
            public const string Name = "SE Asia Standard Time";
            public const string DisplayName = "Indochina Time (ICT)";
            public const string Abbreviation = "ICT";
            public const int OffsetHours = 7;
        }
    }
}
